vnext/Microsoft.ReactNative.Cxx/TurboModuleProvider.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include <JSI/JsiAbiApi.h>
+#include <JSI/JsiApiContext.h>
+#include <ReactCommon/TurboModule.h>
+#include <winrt/Microsoft.ReactNative.h>
+#include <winrt/Windows.Foundation.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+// CallInvoker on top of JSDispatcher.
+struct AbiCallInvoker : facebook::react::CallInvoker {
+  AbiCallInvoker(IReactDispatcher const &jsDispatcher) : m_jsDispatcher(jsDispatcher) {}
+
+  void invokeAsync(std::function<void()> &&func) override {
```

Comment:
>{ [](start = 58, length = 1)

Could these methods be defined in .cpp file instead of .h?

Reply:
I have moved the non-trivial implementation to a .cpp file.